src/librustc/session/mod.rs

```diff
@@ -176,14 +176,15 @@ impl Session {
     pub fn abort_if_errors(&self) {
         self.diagnostic().abort_if_errors();
     }
-    pub fn abort_if_new_errors<F>(&self, mut f: F)
-        where F: FnMut()
+    pub fn abort_if_new_errors<F, T>(&self, f: F) -> T
```

Comment:
I know this sort of suggestion belonged with PR #30320, but: Can you rename `fn abort_if_new_errors` to make it clearer what its relationship with the callback is?

In particular, when I see an invocation like `sess.abort_if_new_errors(|| { stuff ... })`, especially if my main experience is with the code base prior to PR #30320, my first thought is "this will somehow abort if we've seen new errors since the last time I invoked it, and it will also call this callback as part of the aborting process."

(That is of course not what is happening at all.)

The function you have here is more like a `try_and_abort_on_new_errors(|| { stuff ... })`, which I admit is verbose.

Anyway I won't r- the PR on this suggestion, but I would appreciate if you consider the idea.